java-checks/src/main/java/org/sonar/java/checks/CognitiveComplexityMethodCheck.java

```diff
@@ -116,10 +118,12 @@ private void increaseComplexityByOne(Tree tree) {
     private void increaseComplexity(Tree tree, int increase) {
       complexity += increase;
       if(!ignored.contains(tree)) {
```

Comment:
I might be wrong but wouldn't adding the succeeding if tree to the ignored set achieve the purpose simplier ?